File: src/models/notebook-absence.ts
export type NotebookAbsence = Readonly<{
  id: string

  startDate: Date
  endDate: Date

  justified: boolean
  opened: boolean

  daysMissed: number
  hoursMissed: number
  minutesMissed: number

  shouldParentsJustify: boolean
  administrativelyFixed: boolean

  isReasonUnknown: boolean
  reason?: string
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/NotebookAbsence.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class NotebookAbsence(
    val id: String,

    val startDate: LocalDateTime,
    val endDate: LocalDateTime,

    val justified: Boolean,
    val opened: Boolean,

    val daysMissed: Double,
    val hoursMissed: Int,
    val minutesMissed: Int,

    val shouldParentsJustify: Boolean,
    val administrativelyFixed: Boolean,

    val isReasonUnknown: Boolean,
    val reason: String? = null
)